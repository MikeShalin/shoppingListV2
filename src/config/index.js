/** @flow **/
export const key = 'productList'
