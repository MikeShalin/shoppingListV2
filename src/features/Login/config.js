/** @flow **/

export const formFields = {
  login: {
    label: 'First Name',
    placeholder: 'First Name',
  },
  password: {
    label: 'Enter Password',
    placeholder: 'Enter Password',
  },
}