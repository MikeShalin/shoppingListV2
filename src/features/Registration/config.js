/** @flow **/

export const password = {
  label: 'Enter Password',
  placeholder: 'Enter Password',
  name: 'password',
  type: 'password',
}

export const login = {
  label: 'Enter email',
  placeholder: 'Enter email',
  name: 'login',
  type: 'email',
}

export const withTerms = {
  label: 'I agree to the Terms and Conditions',
}

export const message = {
  success: {
    header: 'Registration Completed',
    content: 'You`re all signed up for the newsletter',
  },
  error: {
    header: 'Registration Failed',
    content: 'You can only sign up for an account once with a given e-mail address.',
  },
}
