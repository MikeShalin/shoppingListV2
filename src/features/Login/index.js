/** @flow **/
import * as React from 'react'
import Form from 'Core/Form'
import { formFields } from './config'

const FormExampleForm = () => <Form {...formFields}/>

export default FormExampleForm