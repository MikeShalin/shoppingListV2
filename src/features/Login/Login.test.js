import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import { formFields } from './config'
import Login from './index'

describe('config login component', () => {
  it('formFields matches initial state', () => {
    expect(formFields).toEqual({
      login: {
        label: 'First Name',
        placeholder: 'First Name',
      },
      password: {
        label: 'Enter Password',
        placeholder: 'Enter Password',
      },
    })
  })
})

describe('Login component', () => {
  it('matches early snapshots', () => {
    const tree = shallow(<Login />)
    expect(shallowToJson(tree)).toMatchSnapshot()
  })
})