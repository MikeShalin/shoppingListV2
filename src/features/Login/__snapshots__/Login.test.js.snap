// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component matches early snapshots 1`] = `
<MainForm
  login={
    Object {
      "label": "First Name",
      "placeholder": "First Name",
    }
  }
  password={
    Object {
      "label": "Enter Password",
      "placeholder": "Enter Password",
    }
  }
/>
`;
